### ISSUE TITLE
Warning when batch_words > MAX_WORDS_IN_BATCH in word2vec

### ISSUE THREAD

**Issue by louisabraham** [RaRe-Technologies/gensim#2801]
The doc says

         batch_words : int, optional
            Target size (in words) for batches of examples passed to worker threads (and
            thus cython routines).(Larger batches will be passed if individual
            texts are longer than 10000 words, but the standard cython code truncates to that maximum.)

I think a larger value should not be accepted.

**Comment by piskvorky** [comment]
Good idea. I'd also get rid of the bracket in the docstring at the same time (the `(` and `)`, not the content). It reads weird and is not really needed, this is important information.

@louisabraham Can you open a PR?

**Comment by dsandeep0138** [comment]
@piskvorky I have worked on the changes. Can you please review and let me know if you have any comments? 

Keras related test cases seem to be failing and are unrelated to the changes made. Can you please guide on fixing these?
